### ISSUE TITLE
Directory value in path option for rotate-ca command is getting truncated

### ISSUE THREAD

**Issue by mdrahman-suse** [k3s-io/k3s#6951]
**Environmental Info:**
K3s Version: 

k3s version v1.26.1+k3s-7739c8b9 (7739c8b9)
go version go1.19.5

Node(s) CPU architecture, OS, and Version: 

Linux 5.15.0-1026-aws #30-Ubuntu SMP
$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.1 LTS"

Cluster Configuration:

1 server, 1 agent

**Describe the bug:**

When the directory value in path option is provided in rotate-ca command, the value is getting truncated at the end.
For example: 
```
$ sudo k3s certificate rotate-ca --path=/this/path/is/invalid
WARN[0000] failed to read /this/path/is/tls/etcd/server-ca.key
WARN[0000] failed to read /this/path/is/tls/etcd/peer-ca.crt
WARN[0000] failed to read /this/path/is/tls/etcd/peer-ca.key
WARN[0000] failed to read /this/path/is/tls/client-ca.key
WARN[0000] failed to read /this/path/is/tls/request-header-ca.key
WARN[0000] failed to read /this/path/is/tls/server-ca.key
WARN[0000] failed to read /this/path/is/tls/etcd/server-ca.crt
WARN[0000] failed to read /this/path/is/tls/server-ca.crt
WARN[0000] failed to read /this/path/is/tls/service.key
WARN[0000] failed to read /this/path/is/tls/request-header-ca.crt
WARN[0000] failed to read /this/path/is/tls/client-ca.crt
FATA[0000] see server log for details: https://127.0.0.1:6443/v1-k3s/cert/cacerts?force=false: 500 Internal Server Error certificate error ID 41847
```
The path is read as `/this/path/is` instead of `/this/path/is/invalid`

**Steps To Reproduce:**

- Installed K3s: `curl -sfL https://get.k3s.io | sudo INSTALL_K3S_COMMIT=7739c8b97ee453a82d2762f2c2172c64e4aaa825 sh -`

**Expected behavior:**

The provided path should be read properly

**Actual behavior:**

The provided path is read as truncated

**Additional context / logs:**

**Comment by brandond** [comment]
Note: This works if the path has a trailing slash. `k3s certificate rotate-ca --path=/var/lib/rancher/k3s/server/` works but `k3s certificate rotate-ca --path=/var/lib/rancher/k3s/server` does not.

**Comment by VestigeJ** [comment]
#Environment Details
reproduced original issue cited using COMMIT=7739c8b97ee453a82d2762f2c2172c64e4aaa825
validated fix using COMMIT=21560155217bdfc12c5105d7cb79fad61b2985aa

*Infrastructure*
- [X] Cloud

*Node(s) CPU architecture, OS, and version:*

```
Linux 5.14.21-150400.24.11-default x86_64 GNU/Linux 
PRETTY_NAME="SUSE Linux Enterprise Server 15 SP4"
```

*Cluster Configuration:*

```
NAME               STATUS   ROLES                  AGE   VERSION
ip-bricks          Ready    control-plane,master   22m   v1.26.1+k3s-21560155
ip-mortar          Ready    etcd                   22m   v1.26.1+k3s-21560155
ip-pizza-oven      Ready    etcd                   22m   v1.26.1+k3s-21560155
ip-dinner          Ready    etcd                   22m   v1.26.1+k3s-21560155 
```

*Config.yaml:*

```
disable-etcd: true
node-taint:
  - node-role.kubernetes.io/control-plane:NoSchedule //note the above section wasn't required but I'm using this cluster for testing multiple issues in tandem
write-kubeconfig-mode: 644
debug: true
token: millionsofpeaches
profile: cis-1.23
selinux: true
server: https://ETCD_SERVER_IP:6443
```

<details>
<summmary><h4> Reproduced the original issue using the COMMIT provided above from MD </h4></summary>



```
 curl https://get.k3s.io --output install-"k3s".sh
 sudo chmod +x install-"k3s".sh
 sudo groupadd --system etcd && sudo useradd -s /sbin/nologin --system -g etcd etcd
 sudo modprobe ip_vs_rr
 sudo modprobe ip_vs_wrr
 sudo modprobe ip_vs_sh
 sudo printf "on_oovm.panic_on_oom=0 \nvm.overcommit_memory=1 \nkernel.panic=10 \nkernel.panic_ps=1 \nkernel.panic_on_oops=1 \n" > ~/90-kubelet.conf
 sudo cp 90-kubelet.conf /etc/sysctl.d/
 sudo systemctl restart systemd-sysctl
$ conf //vim /etc/rancher/k3s/config.yaml
$ COMMIT=7739c8b97ee453a82d2762f2c2172c64e4aaa825
$ sudo INSTALL_K3S_COMMIT=$COMMIT INSTALL_K3S_EXEC=server ./install-k3s.sh
$ set_kubefig //KUBECONFIG=/etc/rancher/k3s/k3s.yaml
$ kgn //kubectl get nodes
$ kgp -A //kubectl get pods -A
$ sudo k3s certificate rotate-ca --path=/this/path/is/looooong
$ sudo k3s certificate rotate-ca --path=/this/path/is/invalid/
$ sudo k3s certificate rotate-ca --path=/this/path/is/
$ sudo k3s certificate rotate-ca --path=/this/path/is/loooonger/than/usual/because/qa/
$ sudo k3s certificate rotate-ca --path=/this/p
$ sudo k3s certificate rotate-ca --path=/this/path/
$ k3s -v 
```


**Results:**

$ sudo k3s certificate rotate-ca --path=/this/path/is/looooong
```
WARN[0000] failed to read /this/path/is/tls/server-ca.crt 
WARN[0000] failed to read /this/path/is/tls/etcd/peer-ca.key 
WARN[0000] failed to read /this/path/is/tls/etcd/peer-ca.crt 
WARN[0000] failed to read /this/path/is/tls/server-ca.key 
WARN[0000] failed to read /this/path/is/tls/request-header-ca.key 
WARN[0000] failed to read /this/path/is/tls/etcd/server-ca.key 
WARN[0000] failed to read /this/path/is/tls/client-ca.crt 
WARN[0000] failed to read /this/path/is/tls/service.key 
WARN[0000] failed to read /this/path/is/tls/etcd/server-ca.crt 
WARN[0000] failed to read /this/path/is/tls/request-header-ca.crt 
WARN[0000] failed to read /this/path/is/tls/client-ca.key 
FATA[0000] see server log for details: https://127.0.0.1:6443/v1-k3s/cert/cacerts?force=false: 500 Internal Server Error certificate error ID 90559 
```

$ sudo k3s certificate rotate-ca --path=/this/path/is/looooong/
```
WARN[0000] failed to read /this/path/is/looooong/tls/etcd/server-ca.crt 
WARN[0000] failed to read /this/path/is/looooong/tls/etcd/server-ca.key 
WARN[0000] failed to read /this/path/is/looooong/tls/etcd/peer-ca.key 
WARN[0000] failed to read /this/path/is/looooong/tls/server-ca.key 
WARN[0000] failed to read /this/path/is/looooong/tls/service.key 
WARN[0000] failed to read /this/path/is/looooong/tls/request-header-ca.crt 
WARN[0000] failed to read /this/path/is/looooong/tls/request-header-ca.key 
WARN[0000] failed to read /this/path/is/looooong/tls/server-ca.crt 
WARN[0000] failed to read /this/path/is/looooong/tls/client-ca.key 
WARN[0000] failed to read /this/path/is/looooong/tls/etcd/peer-ca.crt 
WARN[0000] failed to read /this/path/is/looooong/tls/client-ca.crt 
FATA[0000] see server log for details: https://127.0.0.1:6443/v1-k3s/cert/cacerts?force=false: 500 Internal Server Error certificate error ID 01052 
```

$ sudo k3s certificate rotate-ca --path=/this/path/is/invalid/
```
WARN[0000] failed to read /this/path/is/invalid/tls/request-header-ca.crt 
WARN[0000] failed to read /this/path/is/invalid/tls/etcd/server-ca.key 
WARN[0000] failed to read /this/path/is/invalid/tls/etcd/peer-ca.crt 
WARN[0000] failed to read /this/path/is/invalid/tls/etcd/peer-ca.key 
WARN[0000] failed to read /this/path/is/invalid/tls/service.key 
WARN[0000] failed to read /this/path/is/invalid/tls/server-ca.crt 
WARN[0000] failed to read /this/path/is/invalid/tls/server-ca.key 
WARN[0000] failed to read /this/path/is/invalid/tls/client-ca.crt 
WARN[0000] failed to read /this/path/is/invalid/tls/client-ca.key 
WARN[0000] failed to read /this/path/is/invalid/tls/request-header-ca.key 
WARN[0000] failed to read /this/path/is/invalid/tls/etcd/server-ca.crt 
FATA[0000] see server log for details: https://127.0.0.1:6443/v1-k3s/cert/cacerts?force=false: 500 Internal Server Error certificate error ID 16966 
```

$ sudo k3s certificate rotate-ca --path=/this/path/is/
```
WARN[0000] failed to read /this/path/is/tls/server-ca.key 
WARN[0000] failed to read /this/path/is/tls/etcd/server-ca.crt 
WARN[0000] failed to read /this/path/is/tls/etcd/peer-ca.crt 
WARN[0000] failed to read /this/path/is/tls/etcd/peer-ca.key 
WARN[0000] failed to read /this/path/is/tls/request-header-ca.key 
WARN[0000] failed to read /this/path/is/tls/server-ca.crt 
WARN[0000] failed to read /this/path/is/tls/client-ca.key 
WARN[0000] failed to read /this/path/is/tls/service.key 
WARN[0000] failed to read /this/path/is/tls/etcd/server-ca.key 
WARN[0000] failed to read /this/path/is/tls/client-ca.crt 
WARN[0000] failed to read /this/path/is/tls/request-header-ca.crt 
FATA[0000] see server log for details: https://127.0.0.1:6443/v1-k3s/cert/cacerts?force=false: 500 Internal Server Error certificate error ID 86186 
```

$ sudo k3s certificate rotate-ca --path=/this/path/is
```
WARN[0000] failed to read /this/path/tls/etcd/server-ca.crt 
WARN[0000] failed to read /this/path/tls/etcd/server-ca.key 
WARN[0000] failed to read /this/path/tls/server-ca.key  
WARN[0000] failed to read /this/path/tls/request-header-ca.crt 
WARN[0000] failed to read /this/path/tls/etcd/peer-ca.key 
WARN[0000] failed to read /this/path/tls/service.key    
WARN[0000] failed to read /this/path/tls/client-ca.crt  
WARN[0000] failed to read /this/path/tls/client-ca.key  
WARN[0000] failed to read /this/path/tls/etcd/peer-ca.crt 
WARN[0000] failed to read /this/path/tls/server-ca.crt  
WARN[0000] failed to read /this/path/tls/request-header-ca.key 
FATA[0000] see server log for details: https://127.0.0.1:6443/v1-k3s/cert/cacerts?force=false: 500 Internal Server Error certificate error ID 33250 
```

$ sudo k3s certificate rotate-ca --path=/this/is/a/path/to/nowhere
```
WARN[0000] failed to read /this/is/a/path/to/tls/server-ca.key 
WARN[0000] failed to read /this/is/a/path/to/tls/request-header-ca.key 
WARN[0000] failed to read /this/is/a/path/to/tls/etcd/server-ca.crt 
WARN[0000] failed to read /this/is/a/path/to/tls/etcd/peer-ca.key 
WARN[0000] failed to read /this/is/a/path/to/tls/server-ca.crt 
WARN[0000] failed to read /this/is/a/path/to/tls/service.key 
WARN[0000] failed to read /this/is/a/path/to/tls/request-header-ca.crt 
WARN[0000] failed to read /this/is/a/path/to/tls/etcd/server-ca.key 
WARN[0000] failed to read /this/is/a/path/to/tls/etcd/peer-ca.crt 
WARN[0000] failed to read /this/is/a/path/to/tls/client-ca.crt 
WARN[0000] failed to read /this/is/a/path/to/tls/client-ca.key 
FATA[0000] see server log for details: https://127.0.0.1:6443/v1-k3s/cert/cacerts?force=false: 500 Internal Server Error certificate error ID 60215
```
</details>

<details>
<summmary><h4> Validated fix using latest COMMIT </h4></summary>
 
## Validation Steps
 
```
 curl https://get.k3s.io --output install-"k3s".sh
 sudo chmod +x install-"k3s".sh
 sudo groupadd --system etcd && sudo useradd -s /sbin/nologin --system -g etcd etcd
 sudo modprobe ip_vs_rr
 sudo modprobe ip_vs_wrr
 sudo modprobe ip_vs_sh
 sudo printf "on_oovm.panic_on_oom=0 \nvm.overcommit_memory=1 \nkernel.panic=10 \nkernel.panic_ps=1 \nkernel.panic_on_oops=1 \n" > ~/90-kubelet.conf
 sudo cp 90-kubelet.conf /etc/sysctl.d/
 sudo systemctl restart systemd-sysctl
$ conf //vim /etc/rancher/k3s/config.yaml
$ COMMIT=21560155217bdfc12c5105d7cb79fad61b2985aa
$ sudo INSTALL_K3S_COMMIT=$COMMIT INSTALL_K3S_EXEC=server ./install-k3s.sh
$ set_kubefig //KUBECONFIG=/etc/rancher/k3s/k3s.yaml
$ kgn //kubectl get nodes
$ kgp -A //kubectl get pods -A
$ sudo k3s certificate rotate-ca --path=/this/path/is/looooong
$ sudo k3s certificate rotate-ca --path=/this/path/is/invalid/
$ sudo k3s certificate rotate-ca --path=/this/path/is/
$ sudo k3s certificate rotate-ca --path=/this/path/is/loooonger/than/usual/because/qa/
$ sudo k3s certificate rotate-ca --path=/this/p
$ sudo k3s certificate rotate-ca --path=/this/path/
$ k3s -v 
$ get_report //produce this template
``` 
 
**Additional context / logs:**
Tested various path lengths

$ sudo k3s certificate rotate-ca --path=/this/path/is/looooong
```
WARN[0000] failed to read /this/path/is/looooong/tls/etcd/peer-ca.crt 
WARN[0000] failed to read /this/path/is/looooong/tls/request-header-ca.key 
WARN[0000] failed to read /this/path/is/looooong/tls/etcd/server-ca.key 
WARN[0000] failed to read /this/path/is/looooong/tls/etcd/peer-ca.key 
WARN[0000] failed to read /this/path/is/looooong/tls/server-ca.key 
WARN[0000] failed to read /this/path/is/looooong/tls/client-ca.crt 
WARN[0000] failed to read /this/path/is/looooong/tls/service.key 
WARN[0000] failed to read /this/path/is/looooong/tls/etcd/server-ca.crt 
WARN[0000] failed to read /this/path/is/looooong/tls/client-ca.key 
WARN[0000] failed to read /this/path/is/looooong/tls/request-header-ca.crt 
WARN[0000] failed to read /this/path/is/looooong/tls/server-ca.crt 
FATA[0000] see server log for details: https://127.0.0.1:6443/v1-k3s/cert/cacerts?force=false: 500 Internal Server Error certificate error ID 01294 
```

$ sudo k3s certificate rotate-ca --path=/this/path/is/invalid/
```
WARN[0000] failed to read /this/path/is/invalid/tls/client-ca.key 
WARN[0000] failed to read /this/path/is/invalid/tls/etcd/server-ca.crt 
WARN[0000] failed to read /this/path/is/invalid/tls/server-ca.key 
WARN[0000] failed to read /this/path/is/invalid/tls/client-ca.crt 
WARN[0000] failed to read /this/path/is/invalid/tls/etcd/peer-ca.key 
WARN[0000] failed to read /this/path/is/invalid/tls/etcd/peer-ca.crt 
WARN[0000] failed to read /this/path/is/invalid/tls/service.key 
WARN[0000] failed to read /this/path/is/invalid/tls/request-header-ca.key 
WARN[0000] failed to read /this/path/is/invalid/tls/etcd/server-ca.key 
WARN[0000] failed to read /this/path/is/invalid/tls/request-header-ca.crt 
WARN[0000] failed to read /this/path/is/invalid/tls/server-ca.crt 
FATA[0000] see server log for details: https://127.0.0.1:6443/v1-k3s/cert/cacerts?force=false: 500 Internal Server Error certificate error ID 73556 
```

$ sudo k3s certificate rotate-ca --path=/this/path/is/
```
WARN[0000] failed to read /this/path/is/tls/client-ca.key 
WARN[0000] failed to read /this/path/is/tls/client-ca.crt 
WARN[0000] failed to read /this/path/is/tls/service.key 
WARN[0000] failed to read /this/path/is/tls/request-header-ca.key 
WARN[0000] failed to read /this/path/is/tls/etcd/peer-ca.crt 
WARN[0000] failed to read /this/path/is/tls/server-ca.crt 
WARN[0000] failed to read /this/path/is/tls/request-header-ca.crt 
WARN[0000] failed to read /this/path/is/tls/etcd/server-ca.crt 
WARN[0000] failed to read /this/path/is/tls/etcd/server-ca.key 
WARN[0000] failed to read /this/path/is/tls/etcd/peer-ca.key 
WARN[0000] failed to read /this/path/is/tls/server-ca.key 
FATA[0000] see server log for details: https://127.0.0.1:6443/v1-k3s/cert/cacerts?force=false: 500 Internal Server Error certificate error ID 60346 
```

$ sudo k3s certificate rotate-ca --
```
path=/this/path/is/loooonger/than/usual/because/qa/
WARN[0000] failed to read /this/path/is/loooonger/than/usual/because/qa/tls/etcd/server-ca.key 
WARN[0000] failed to read /this/path/is/loooonger/than/usual/because/qa/tls/service.key 
WARN[0000] failed to read /this/path/is/loooonger/than/usual/because/qa/tls/etcd/server-ca.crt 
WARN[0000] failed to read /this/path/is/loooonger/than/usual/because/qa/tls/request-header-ca.crt 
WARN[0000] failed to read /this/path/is/loooonger/than/usual/because/qa/tls/request-header-ca.key 
WARN[0000] failed to read /this/path/is/loooonger/than/usual/because/qa/tls/client-ca.key 
WARN[0000] failed to read /this/path/is/loooonger/than/usual/because/qa/tls/etcd/peer-ca.key 
WARN[0000] failed to read /this/path/is/loooonger/than/usual/because/qa/tls/etcd/peer-ca.crt 
WARN[0000] failed to read /this/path/is/loooonger/than/usual/because/qa/tls/server-ca.key 
WARN[0000] failed to read /this/path/is/loooonger/than/usual/because/qa/tls/client-ca.crt 
WARN[0000] failed to read /this/path/is/loooonger/than/usual/because/qa/tls/server-ca.crt 
FATA[0000] see server log for details: https://127.0.0.1:6443/v1-k3s/cert/cacerts?force=false: 500 Internal Server Error certificate error ID 97595 
```

$ sudo k3s certificate rotate-ca --path=/this/p
```
WARN[0000] failed to read /this/p/tls/client-ca.key     
WARN[0000] failed to read /this/p/tls/client-ca.crt     
WARN[0000] failed to read /this/p/tls/etcd/server-ca.crt 
WARN[0000] failed to read /this/p/tls/etcd/peer-ca.key  
WARN[0000] failed to read /this/p/tls/server-ca.key     
WARN[0000] failed to read /this/p/tls/server-ca.crt     
WARN[0000] failed to read /this/p/tls/service.key       
WARN[0000] failed to read /this/p/tls/request-header-ca.crt 
WARN[0000] failed to read /this/p/tls/request-header-ca.key 
WARN[0000] failed to read /this/p/tls/etcd/server-ca.key 
WARN[0000] failed to read /this/p/tls/etcd/peer-ca.crt  
FATA[0000] see server log for details: https://127.0.0.1:6443/v1-k3s/cert/cacerts?force=false: 500 Internal Server Error certificate error ID 59732 
```

$ sudo k3s certificate rotate-ca --path=/this/path/
```
WARN[0000] failed to read /this/path/tls/etcd/server-ca.crt 
WARN[0000] failed to read /this/path/tls/server-ca.key  
WARN[0000] failed to read /this/path/tls/request-header-ca.crt 
WARN[0000] failed to read /this/path/tls/etcd/peer-ca.crt 
WARN[0000] failed to read /this/path/tls/server-ca.crt  
WARN[0000] failed to read /this/path/tls/client-ca.crt  
WARN[0000] failed to read /this/path/tls/request-header-ca.key 
WARN[0000] failed to read /this/path/tls/etcd/server-ca.key 
WARN[0000] failed to read /this/path/tls/client-ca.key  
WARN[0000] failed to read /this/path/tls/service.key    
WARN[0000] failed to read /this/path/tls/etcd/peer-ca.key 
FATA[0000] see server log for details: https://127.0.0.1:6443/v1-k3s/cert/cacerts?force=false: 500 Internal Server Error certificate error ID 06028 
```

</details>
